addons/pos_coupon/static/src/js/coupon.js

```diff
@@ -1017,20 +1017,33 @@ odoo.define('pos_coupon.pos', function (require) {
                 .join(',');
         },
         _createDiscountRewards: function (program, coupon_id, amountsToDiscount) {
-            const discountRewards = Object.entries(amountsToDiscount).map(([tax_keys, amount]) => {
+            const rewards = [];
+            let totalDiscount = 0;
+            const totalAmountsToDiscount = Object.values(amountsToDiscount).reduce((a, b) => a + b, 0);
+            for (let [tax_keys, amount] of Object.entries(amountsToDiscount)) {
                 let discountAmount = (amount * program.discount_percentage) / 100.0;
+                let maxDiscount = amount / totalAmountsToDiscount * program.discount_max_amount || Infinity;
+                discountAmount = Math.min(discountAmount, maxDiscount);
+                if (totalDiscount >= program.discount_max_amount && program.discount_max_amount > 0) {
```

Comment:
you can remove this condition. we should not be able to meet this condition, and even if we meet it, it means that we'll have a bug and we'll be hiding it